modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -421,66 +421,142 @@ public PathElement[] getCaretShape(int offset, boolean isLeading,
     }
 
     @Override
-    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart) {
+    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart, boolean forTextFlow) {
```

Comment:
I did not find splitting the `getHitInfo` method based on some condition very helpful. Since we are calculating effective value of x and TextRun, many values are dependant on intermediate values and the calculation involves many variables.  Hence I decided to keep the major calculations inside `getHitInfo` method. 
Added 2 methods to move the calculation of values like previous line text width and width of LTR text in RTL text out of `getHitInfo`. 

> You can also reduce the nesting of the first `if/else` by returning early:
> 
Made changes to return early in this case.